cmd/diagnostic/PyTorch.go

```diff
@@ -0,0 +1,171 @@
+package diagnostic
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+func getPodMachineID(podID, apiKey string) string {
+	url := fmt.Sprintf("https://api.runpod.io/graphql?api_key=%s", apiKey)
+	headers := map[string]string{
+		"Content-Type": "application/json",
+	}
+	query := `
+        query Pod($podId: String!) {
+          pod(input: { podId: $podId }) {
+            machineId
+          }
+        }
+    `
+	data := map[string]interface{}{
```

Comment:
MINOR: General go HTTP/REST performance tips:
- use a struct here rather than `map[string]any` to avoid extraneous allocations.
- don't make a map to iterate through the headers: just do `req.Header.Set("Content-Type", "application/json")`
- don't make a new `http.Client` for each request: they're safe to share between invocations. if you don't care which client to use, use `http.DefaultClient`
- either or both of the client  or request context should have a timeout.